guardian/typerighter
# Add similarity ordering to rule search

## What does this change?

I was searching for a dictionary rule, and realised that Typerighter's rule search is not very good. This PR adds similarity ordering to rule search.

Part of the reason is that we use a trigram index for comparing words, but do not order by similarity to the search term. Added here.

|Before|After|
|--|--|
|<img width="908" alt="Screenshot 2024-04-08 at 09 40 25" src="https://github.com/guardian/typerighter/assets/7767575/89102f30-bb24-4d4c-8ed3-6ac8cb32083f">|<img width="908" alt="Screenshot 2024-04-06 at 15 18 48" src="https://github.com/guardian/typerighter/assets/7767575/43a761eb-27b4-4bbc-aa6b-83c0594262ee">|

## How to test

- The automated tests should pass. Existing ones have reasonable coverage of search functionality. There's a new one for relevance.
- Running locally or in CODE, observe the behaviour of search. It should favour text that most closely matches the rule type, pattern, description, or replacement.
